pallets/marketplace/src/benchmarking.rs

```diff
@@ -0,0 +1,136 @@
+#![cfg(feature = "runtime-benchmarks")]
+#![allow(clippy::useless_conversion)]
+
+use super::*;
+
+use crate::Pallet as Marketplace;
+use frame_benchmarking::{account, benchmarks, vec};
+use frame_system::RawOrigin;
+use sp_runtime::{traits::UniqueSaturatedInto, SaturatedConversion};
+use sp_std::convert::TryInto;
+
+const SEED: u32 = 0;
+const ENDOWMENT: u32 = 1_000_000;
+
+macro_rules! bvec {
+	($( $x:tt )*) => {
+		vec![$( $x )*].try_into().unwrap()
+	}
+}
+
+fn create_account<T: Config>(name: &'static str, index: u32) -> T::AccountId {
+	let caller: T::AccountId = account(name, index, SEED);
+
+	let amount = unit(ENDOWMENT);
+	<T as pallet_nft::Config>::Currency::deposit_creating(&caller, amount.unique_saturated_into());
+
+	caller
+}
+
+fn unit(d: u32) -> u128 {
+	let d: u128 = d.into();
+	d.saturating_mul(1_000_000_000_000)
+}
+
+benchmarks! {
+	buy {
+		let caller = create_account::<T>("caller", 0);
+		let caller2 = create_account::<T>("caller2", 0);
+		pallet_nft::Pallet::<T>::create_class(RawOrigin::Signed(caller.clone()).into(), ClassType::Marketplace, bvec![0])?;
+		pallet_nft::Pallet::<T>::mint(RawOrigin::Signed(caller.clone()).into(), 0u32.into(), Some(caller.clone()), Some(20), Some(b"metadata".to_vec()))?;
```

Comment:
Benchmarking : I believe this is metadata should be the longest possible to benchmark the worst case. 

I would just create the longest metadata and use it each benchmark where metadata is used

Reply:
yes agreed, but it is possible that even some callable functions from pallet will be removed and some will appear so let's keep benchmarking as is and refactor as the last thing, will keep that on my list